use logger created from logger factory instead of console logger (#249)
elastic/apm-agent-dotnet: docs/configuration.asciidoc

NOTE: By simply calling `app.UseElasticApm()` without the overload, the agent will read configurations only from environment variables.

> NOTE: For better user experience, `LogLevel` for internal apm logger is used from standard `Logging` section inside `appsettings.json` with `Apm` category name.

Reviewer: These changes only apply to ASP.NET Core (since we activate the new logger in the `ApmMiddlewareExtension`), so I'd try to make it clear in the docs.

I propose to add this hint to the beginning: `For better user experience in case of ASP.NET Core, `LogLevel` for internal apm …`

Author: added

> This is a typical `appsettings.json` file that contains some sample configuration. The part below `ElasticApm` is fetched by the agent if the corresponding `IConfiguration` is passed to the agent.

Reviewer: Similarly, here I'd make clear that this is an `appsettings.json` sample for ASP.NET Core.

Author: added

[source,js]
----
{
  "Logging": {
    "LogLevel": {
      "Default": "Warning",
      "Apm": "Debug"
    }
  },
  "AllowedHosts": "*",
  "ElasticApm":

>     {

Reviewer: Additionally I'd add another block below this one to explain how to use `ElasticApm:LogLevel` when ASP.NET Core is not involved. 

Something like this:

In case you don't activate the agent with the `UseElasticApm()` method (typically non-ASP.NET Core scenario) then you can set the loglevel of the agent with the `ElasticApm:LogLevel` setting, as shown in the following `appsettings.json` file:

[and then use the original file]

(_feel free to adapt or improve the text if you feel something is not ok_)

Author: added

NOTE: By simply calling `app.UseElasticApm()` without the overload, the agent will read configurations only from environment variables.

> NOTE: For better user experience in case of ASP.NET Core, `LogLevel` for internal apm logger is used from standard `Logging` section inside `appsettings.json` with `Apm` category name.

Reviewer 2: I don't love the idea of having two _NOTEs_ in a row:
<img width="784" alt="Screen Shot 2019-07-01 at 2 17 11 PM" src="https://user-images.githubusercontent.com/5618806/60467248-f2493200-9c0a-11e9-9b91-21ee18cd43b6.png">

Does this new information need to be in a note? Also, it could be lack of .NET knowledge, but I'm not exactly sure what this is trying to say? @gregkalapos do you?

Reviewer: Yes, I do. This changes how to set the loglevel for the agent. In case of ASP.NET Core, here is how to do it:
```
{
	  "Logging": {
	    "LogLevel": {
	      "Default": "Warning",
	      "ElasticApm": "Debug"
	    }
	  },
	  "AllowedHosts": "*",
	  "ElasticApm":
	    {
	      "ServerUrls":  "http://myapmserver:8200",
	      "TransactionSampleRate": 1.0
	    }
	}
```

The `Logging` part is standard and also used by the ASP.NET Core logger - so we (the APM Agent) log to the same logger.

Now, if you don't use ASP.NET Core (e.g. you have a command line application) we can't register our logger with the default logger (since there isn't really one), so in that case here is how you set the log level (it's moved to the `ElasticApm` section:

```
{
	  "Logging": {
	    "LogLevel": {
	      "Default": "Warning",
	    }
	  },
	  "AllowedHosts": "*",
	  "ElasticApm":
	    {
	      "ServerUrls":  "http://myapmserver:8200",
	      "TransactionSampleRate": 1.0,
              "LogLevel" : "Debug"
	    }
	}
```

Probably there is a better way to phrase this; happy to take suggestions :)

Reviewer: Btw. now that I see the text again, we should change 
> `appsettings.json` with `Apm` category name 

 to 

> `appsettings.json` with `ElasticApm` category name` 

We changed that after one of my comments.

Reviewer 2: This definitely helps, thanks Greg! I'm wondering if we should put this under a "sample config" section... here's what I'm thinking:

```asciidoc
[float]
[[sample-config]]
==== Sample configuration file

Below is a sample `appsettings.json` configuration file for a typical ASP.NET Core application. There are two important takeaways:
1. The part below `ElasticApm` is fetched by the agent if the corresponding `IConfiguration` is passed to the agent.
2. With ASP.NET Core, we recommend setting `LogLevel for the internal APM logger in the standard `Logging` section with the `ElasticApm` category name.

[source,js]
----
{
  "Logging": {
    "LogLevel": {
      "Default": "Warning",
      "ElasticApm": "Debug"
    }
  },
  "AllowedHosts": "*",
  "ElasticApm":
    {
      "ServerUrls":  "http://myapmserver:8200",
      "TransactionSampleRate": 1.0
    }
}
----

In certain scenarios -- typically when you're not using ASP.NET Core -- you may choose not to active the agent with the `UseElasticApm()` method.
In this case, you can set the log level of the agent with `ElasticApm:LogLevel`, as shown in the following `appsettings.json` file:

[source,js]
----
{
  "Logging": {
    "LogLevel": {
      "Default": "Warning"
    }
  },
  "AllowedHosts": "*",
  "ElasticApm":
    {
      "LogLevel":  "Debug",
      "ServerUrls":  "http://myapmserver:8200",
      "TransactionSampleRate": 1.0
    }
}
----
```

Reviewer: Yeah, I think the structure is very good! 

Some comments:

> With ASP.NET Core, we recommend setting `LogLevel` for the internal APM logger in the standard `Logging`

It's not a recommendation, it's the only way to set it, if you do it the other way (like the 2. example above, it'll be ignored). 

> In certain scenarios -- typically when you're not using ASP.NET Core -- you may choose not to active the agent with the `UseElasticApm()` method.

This is very good, I'd just change a small thing: the part with "you may choose not to active the agent with..."  should be a different. It's not really a choice, if you are not on ASP.NET Core you can't really call the `UseElasticApm` method, so it should be something like "you typically don't activate the agent with..."

Overall I think with this structure the doc is good and easy to understand. On the other hand I'm still not sure this is a good move, it's still hard to explain this, it can be confusing. I think it'd be better to just use the same in both cases. Unfortunately I think there is no way to fetch the loglevel from `"ElasticApm":  { "LogLevel":  "..."}` when we use `ILoggerFactory`.

So I see 2+1 options:

- We go with the improved docs 
- We continue thinking about this
- (and +1: we go with the improved docs and reiterate on this and see if there is a better way to always configure the loglevel the same way)

I'd choose the last one. 

@vhatsura let us know what you think.

Reviewer 2: Thanks again Greg! I've added your recommendations and updated the docs below. If you and @vhatsura revisit this and are able to fetch the loglevel in the same place, let me know and I can update the docs accordingly.

```asciidoc
[float]
[[sample-config]]
==== Sample configuration file

Below is a sample `appsettings.json` configuration file for a typical ASP.NET Core application. There are two important takeaways:
1. The part below `ElasticApm` is fetched by the agent if the corresponding `IConfiguration` is passed to the agent.
2. With ASP.NET Core, you must set `LogLevel for the internal APM logger in the standard `Logging` section with the `ElasticApm` category name.

[source,js]
----
{
  "Logging": {
    "LogLevel": {
      "Default": "Warning",
      "ElasticApm": "Debug"
    }
  },
  "AllowedHosts": "*",
  "ElasticApm":
    {
      "ServerUrls":  "http://myapmserver:8200",
      "TransactionSampleRate": 1.0
    }
}
----

In certain scenarios -- like when you're not using ASP.NET Core -- you wont activate the agent with the `UseElasticApm()` method.
In this case, you can set the log level of the agent with `ElasticApm:LogLevel`, as shown in the following `appsettings.json` file:

[source,js]
----
{
  "Logging": {
    "LogLevel": {
      "Default": "Warning"
    }
  },
  "AllowedHosts": "*",
  "ElasticApm":
    {
      "LogLevel":  "Debug",
      "ServerUrls":  "http://myapmserver:8200",
      "TransactionSampleRate": 1.0
    }
}
----
```

Reviewer: Thanks @bmorelli25. I think this is a fairly important decision - I thought about it and opened https://github.com/elastic/apm-agent-dotnet/issues/338 with a suggestion - we'll decide there. Will ping you once we came up with something.

Author: updated

> This is a typical `appsettings.json` file for ASP.NET Core application that contains some sample configuration. The part below `ElasticApm` is fetched by the agent if the corresponding `IConfiguration` is passed to the agent.

Reviewer 2: ```suggestion
Below is a sample `appsettings.json` configuration file for a typical ASP.NET Core application. The part below `ElasticApm` is fetched by the agent if the corresponding `IConfiguration` is passed to the agent.
```

[source,js]
----
{
  "Logging": {
    "LogLevel": {
      "Default": "Warning",
      "ElasticApm": "Debug"
    }
  },
  "AllowedHosts": "*",
  "ElasticApm":
    {
      "ServerUrls":  "http://myapmserver:8200",
      "TransactionSampleRate": 1.0
    }
}
----

> In case you don't activate the agent with the `UseElasticApm()`` method (typically non-ASP.NET Core scenario) then you can set the log level of the agent with the `ElasticApm:LogLevel` setting, as shown in the following `appsettings.json` file:

Reviewer 2: ```suggestion
In certain scenarios -- typically when you're not using ASP.NET Core -- you may choose not to active the agent with the `UseElasticApm()` method.
In this case, you can set the log level of the agent with `ElasticApm:LogLevel`, as shown in the following `appsettings.json` file:
```

NOTE: By simply calling `app.UseElasticApm()` without the overload, the agent will read configurations only from environment variables.

[float]
[[sample-config]]
==== Sample configuration file

Below is a sample `appsettings.json` configuration file for a typical ASP.NET Core application. There are two important takeaways:

> 1. The part below `ElasticApm` is fetched by the agent if the corresponding `IConfiguration` is passed to the agent.

Reviewer 2: ```suggestion

1. The part below `ElasticApm` is fetched by the agent if the corresponding `IConfiguration` is passed to the agent.
```
